### DISCUSSION TITLE
Why is the development team ignoring global issues?

### DISCUSSION THREAD

**Discussion by Deroswent** [medusajs/medusa#9444] [RFC]
First of all, Medusa is a great project and I want to thank all the contributors. 

But it has some very big problems that the developers have not reacted to for years.

I'm talking about two things:

Feature for adding variant specific images (https://github.com/medusajs/medusa/discussions/4456)

and

Multi-language support (https://github.com/medusajs/medusa/discussions/2003)

In today's e-commerce environment, these things are de facto mandatory and most companies can't do without them. And Medusa has not offered anything in this direction for several years.

Worst of all is the uncertainty: the development team is simply silent and does not provide answers to the community's questions: will these features be integrated into Medusa or not?

On behalf of the community, I ask the development team to once and for all provide an answer to the question, are there any plans to implement these features in Medusa?
This question has already been raised many times both in discussions and in Discord, but the answer has not been provided. At the same time, this question is very important when choosing a framework for application development.

I would like to thank the development team in advance for their answers.

**Comment by joekendal** [comment]
I'd imagine the team is hard at work preparing v2.0 and dealing with enterprise clients + their new cloud product. If you want to get an official response it might be easier to post in the Discord.

If they don't respond to #4456 we'll open a PR and get design feedback that way. We are waiting for a schema design before progressing with it. So far, the discussion #9434 has not received any feedback but let's wait a few days and see...

**Comment by AlexAntonides** [comment]
I experience the same, especially for https://github.com/medusajs/medusa/discussions/8682, which for my team is a requirement to continue using Medusa 2.0. I’ve tried to contact the development team on both Github and Discord, but alas. I understand that the Medusa team is working hard for a v2.0 release, but I don’t need the entire solution yet. All I need is an answer whether they’ll keep it in mind for a future release, so I can know in advance whether to drop Medusa 2.0 or not.
At this point we’re at the point of giving up due to the lack of response.

**Comment by olivermrbl** [comment]
@Deroswent @joekendal @AlexAntonides, I apologise deeply for the slow (or nonexistent) response from our side. As Joe pointed out, we are heads down getting everything ready for the official release. I'll get back to these questions over the weekend. 

There is only so much we can do given our small team size, and right now, polishing v2.0 takes highest priority.

**Reply by Deroswent** [reply]
Hi @olivermrbl . First of all, congratulations on the release of Medusa v2. You and the whole development team are great. We love you guys :) And I hope you have a little less work :)

But still, we would like to get some kind of answer to the questions asked earlier in this thread. Judging by the likes in discussions and in discord, these functions are in demand by developers most of all and are necessary in almost any branch of trade.
We are not asking for these features to be implemented immediately, we just want to know if they will be implemented in the foreseeable future?  The lack of answers to these questions from developers discourages many companies from starting work with Medusa.

**Reply by joekendal** [reply]
I spoke with one of the founders about it and they said they have previously discussed internally and decided it wouldn't be a priority.

After implementing our use case, we found that the current Medusa implementation worked better than what we initially proposed. Here's why:

1. We are creating vector embeddings for a recommendation engine and found it's better to create separate products where the style characteristics are unique. And then you just create vector embeddings at the product level. That way you are not creating unecessary vectors and are just pre-filtering by variant size.

2. If Medusa put images at the variant level, we would have a lot of duplicates because it would create a variant for every size and color combo. This could get messy to manage, especially if the images could either be linked across sizes or unique. Difficult for Medusa to generalise the solution for all use cases here.

3. We can just create a custom entity to link products in a way that makes sense for our use case. This is probably a more flexible approach than being forced to handle product relationships in a certain way. Less ideal, but more customisable.

**Reply by jono-allen** [reply]
We have had this problem for a while now on a few projects. We have managed it in a few different ways. The main issue we always had is its complex from a admin management side when products have a large number of product options.
We have found using a cms to manage this works much better and allows for providing variant images on a case by case and using the products image as a fallback.
We have built a plugin for a cms and medusa and hope to share more about it shortly.

**Reply by Deroswent** [reply]
Hi @joekendal  Thanks for your reply and for sharing this information with us.

In my humble opinion, these arguments are true only for developers. But they absolutely do not take into account the requirement of sales industry.
And point 2 looks ridiculous in general.  Probably, the person who made such conclusions is an excellent developer, but has never encountered trading in the real world.  Imagine a garment where one model has 10 different colors.  How do you propose to sell this in a Medusa-based online store if you can't show the customer a photo of what they want to buy? 
Look at any online store - almost all of them post 3-5 photos for each product variant, showing it from all angles. This is the industry standard. This greatly increases sales and improves the user experience, as the customer can take a detailed look at what they wish to buy.  

I realize that from a developer's point of view, such functionality may look complicated. But from a business point of view - it is vital, as it helps to make money. And no business will choose Medusa if it results in lower sales, just because it's difficult for developers to add a feature the business needs.

**Reply by joekendal** [reply]
@Deroswent 

1. Our use case is different to yours. It's not a problem for us we found. Medusa is intended to work for other business use cases besides the one you describe.

2. The problem is something that needs to be decided by Medusa if they're going to find the right technical design. If we have 10 different colors and 5 different angles in 5 different sizes it would create 250 images despite there only being 50 unique images. So this is the problem with adding images to every variant - hassle to manage lots of duplicate images without linking images to specific option values. This could cause problems in the UI and backend implementation if not done with careful consideration and community feedback. So it will take a while for them to ship anything.

3. If the solution was straightforward, we would have opened the PR with the fix already. Instead, we found a better way to implement our specific business use case.

4. It's probably best to not assume you know how Medusa fits in to our specific backend. It mainly serves as a headless foundation for functionalities, but not all.

5. Others are using a headless CMS separately and this is why Medusa did not see it as a priority. I disagree with Medusa's stance though. Medusa should not require an additional CMS or CDN.

**Reply by lukahrobat** [reply]
What about this type of solution https://www.shopify.com/enterprise/blog/combined-listings? This approach combines products and their variants under a single item displayed on the front end.

**Reply by AlexAntonides** [reply]
Is there any news about #8682? We've managed to find a hack to solve this, but this puts the security at risk, so I'd rather have the authentication module actually be used to manage API authentication.

**Comment by meisbellajack** [comment]
I completely feel the frustration here. It’s tough when the core engine is so solid, but these de facto features feel like they're permanently stuck in the backlog. When you're choosing a stack for a client, you can't really tell them to 'wait a few more years' for basic multi-language support.

For those of us who can't wait for the core team to prioritize this, the only real way forward right now is to offload that logic to the frontend or a specialized middleware. I’ve been working on **[Misk Smart Product Options](https://apps.shopify.com/misk-smart-product-options)**, and while it's a Shopify-first tool, it was built specifically to solve this exact 'architectural silence.' It handles the Variant-Specific Images and Conditional Logic on the UI layer so the backend doesn't have to be perfect.

Hopefully, the Medusa team gives us a clear roadmap soon, but in the meantime, layering the logic on the frontend seems to be the only way to stay 'pro' without the technical debt of a custom-coded fix.

**Comment by codeplott** [comment]
Good news! Both features mentioned in this Discussion have already been implemented by the MedusaJS team 💪

- Support for variant images was released in [v2.11.2](https://github.com/medusajs/medusa/releases/tag/v2.11.2)
- Multi-language support was introduced as an experimental feature in [v2.12.3](https://github.com/medusajs/medusa/releases/tag/v2.12.3) and can be enabled using a feature flag in `medusa-config.ts`. You can read more about it here: [Introducing experimental support for Translations](https://medusajs.com/blog/introducing-experimental-translations/)

**Comment by shahednasser** [comment]
Both of the features mentioned in this discussion are now available in Medusa. Make sure to update to the latest version to benefit from them.
